VersionComparisonClass.py

```diff
@@ -0,0 +1,91 @@
+import semantic_version as sv
+import json
+
+class CheckVersion():
+  def __init__(self, VersionRangeList, VersionToCheck):
+    """
+    Inputs 
+    - a list of versions in str format from the file generated from generate_SNYK_vulnerabilities
+      - Example: ["[3.2-alpha,3.3-beta-2)"] 
+      - Splits up by the ','. Hence, every range exists in the following format
+        - ['[3.2-alpha', '3.3-beta-2)']
+    - Version to check in the intervals
+    """
+    self.VersionRangeInput = [interval.strip() for versionRange in VersionRangeList for interval in versionRange.split(',')]
+    self.VersionToCheck = sv.Version.coerce(VersionToCheck)
+
+  def CheckVersionInRange(self):
+    """
+    Iterates through the list in self.VersionRangeInput. 
+    Treats the beginning and end of range as seperate versions and constructs Spec object.
+    Checks the given range in the Spec object. Returns True, if found.
+    Else, goes through the next set of ranges.
+    """
+    VersionRangeList = []
+    for versionPoint in self.VersionRangeInput:
+      VersionRangeList.append(self.UnpackRange(versionPoint))
+      if len(VersionRangeList) == 2:
+        if self.VersionToCheck in sv.Spec(','.join(VersionRangeList)):
+          return True
+        VersionRangeList = []
+    return False
+
+  def UnpackRange(self, versionPoint):
+    """
+    Takes the versions with the corresponding brackets and unpacks it as a mathematical notation
+    This is done to comply with the semantic_version Spec object.
+    """
+    if len(versionPoint) == 1: # [,XXX]
+      if versionPoint[0] == '[' or versionPoint[0] == '(':
+        versionPoint = versionPoint[0] + '0'
+      else:
+        versionPoint = '10000000' + versionPoint[0] #Arbitrarily huge number to satisfy open ended upper limit.
+
+    if versionPoint[0] == '[':
+      return '>=' + str(sv.Version.coerce(versionPoint[1:]))
+    elif versionPoint[0] == '(':
+      return '>' + str(sv.Version.coerce(versionPoint[1:]))
+    elif versionPoint[-1] == ']':
+      return '<=' + str(sv.Version.coerce(versionPoint[:-1]))
+    elif versionPoint[-1] == ')':
+      return '<' + str(sv.Version.coerce(versionPoint[:-1]))
+    else:
+      raise Exception("Range not recongized")
```

Comment:
In this case, let's not raise an Exception. We need to label this cases for manual verification.

Reply:
Got it! See, a4f186f
Thanks!